catkin_tools/context.py

```diff
@@ -75,12 +68,57 @@ class Context(object):
         'blacklist',
     ]
 
-    KEYS = STORED_KEYS + [
+    EXTRA_KEYS = [
         'workspace',
         'profile',
         'space_suffix',
     ]
 
+    KEYS = []
+
+    @classmethod
+    def _create_space_methods(cls, space):
+        def space_abs_getter(self):
+            return getattr(self, '__%s_space_abs' % space)
+
+        def space_getter(self):
+            return getattr(self, '__%s_space' % space)
+
+        def space_setter(self, value):
+            if self.__locked:
+                raise RuntimeError("Setting of context members is not allowed while locked.")
+            setattr(self, '__%s_space' % space, value)
+            setattr(self, '__%s_space_abs' % space, os.path.join(self.__workspace, value))
+
+        def space_exists(self):
+            "Returns true if the space exists"
+            space_abs = getattr(self, '__%s_space_abs' % space)
+            return os.path.exists(space_abs) and os.path.isdir(space_abs)
+
+        setattr(cls, '%s_space' % space, property(space_getter, space_setter))
+        setattr(cls, '%s_space_abs' % space, property(space_abs_getter))
+        setattr(cls, '%s_space_exists' % space, space_exists)
+
+    @classmethod
+    def setup_space_keys(cls):
+        '''
```

Comment:
Fixed.

Potentially this could be called on import, but it felt saner to defer and do it lazily on first use.